05-html-css/flexbox-intro.md

```diff
@@ -0,0 +1,230 @@
+1. More relevant solutions to modern web layouts
+1. Built for responsive design
+1. Promotes thinking about alignment on websites agnostic of language direction
+
+If you choose to not use flexbox as your main way of aligning elements along one dimension, Ada has archived lectures on other ways to do layouts in CSS:
+1. [Layouts using Display](archived/layout-display.md)
+1. [Layouts using Floats](archived/layout-floats.md)
+1. [Layouts using Positioning](archived/layout-positioning.md)
+
+**Note:** Flexbox is fairly new, and adoption for using flexbox isn't widespread. A lot of information and advice on the Internet uses CSS that does not use flexbox, even if flexbox may be a better solution for that problem.
+
+## Rules for Setting Up Flexbox
+
+Flexbox solves the following problem: **How do I define rules for aligning many small elements in one larger container element?** Therefore, to use flexbox, we set up our flexbox with these rules:
+
+1. The "one larger container element" is called **the flex container**. We designate it as the flex container by giving it the CSS rule `display: flex;`
+1. The "many small elements" that we are trying to align are called the flex items. The flex items **must** be and will only be the **direct children** of the flex container
+    - these flex items may have children inside of them. Note: the rules of a flex container do not apply to the children of flex items
+1. The "rules" we want to define for aligning the flex items will be additional properties on the flex container
+
+Note: The flex items of a specific flex container do not have to be all the same kind of element.
+
+### Observe
+
+Let's say we've received this wireframe for a website to create.
+
+![Flexbox Example](./imgs/flexbox-example.png)
+
+Take some time to answer the following questions about that wireframe:
+- What is the flex container?
+- What are the flex items?
+- The elements are ordered alphabetically in the HTML. How are they ordered by appearance?
+- What do you think is the rule about the spacing around the lettered elements?
+
+This is the HTML for the above image:
+
+```html
+<!-- opening html tag, head -->
+<body>
+  <h1>Title</h1>
+  <div class="container">
+    <div> a </div>
+    <div> b </div>
+    <div> c </div>
+    <div> d </div>
+    <div> e </div>
+    <div> f </div>
+    <div> g </div>
+    <div> h </div>
+    <div> i </div>
+    <div> j </div>
+    <div> k </div>
+    <div> l </div>
+    <div> m </div>
+    <div> n </div>
+    <div> o </div>
+    <div> p </div>
+  </div>
+</body>
+<!-- ... -->
+```
+
+Does anything about this surprise you? What parts did you expect?
+
+For the rest of the lecture, we want to continuously ask and observe how flex items:
+
+- align themselves against other items
+- what happens when there are more flex items than can fit in one direction of the flex container? Does it "overflow" into the next space? What is "the next space"?
```

Comment:
The second bullet point is a little awkward. Could it be broken up into multiple points? Maybe

- Align themselves against other items
- Behave when there are more flex items than can fit in one direction
  - How does it overflow into the next space?
  - What is "the next space"?